pandas/core/dtypes/cast.py

```diff
@@ -1183,7 +1205,7 @@ def convert_dtypes(
     return inferred_dtype
 
 
-def maybe_castable(arr) -> bool:
+def maybe_castable(arr: np.ndarray) -> bool:
```

Comment:
we never get EAs here?

Reply:
Atm the only use case is here:
https://github.com/pandas-dev/pandas/blob/126f3cc7e92db0dd92356269d4fb6ea00db30278/pandas/core/construction.py#L538-L557
which according to docstring excludes `ExtensionArray`

Reply:
can you add an assert in the body itself to lock this down. (whether we want it or not is different, but guarantee is good for now)

Reply:
Added